### ISSUE TITLE
Incorrect handling of historic timezone changes

### ISSUE THREAD

**Issue by GULPF** [nim-lang/Nim#6520]
This issue just a more precise description of https://github.com/nim-lang/Nim/issues/4690#issuecomment-318615567, so that #4690 can be closed after https://github.com/nim-lang/Nim/pull/6478 has been merged.

For the C-based backends, `localtime` is used to get the local time representation. `localtime` respects historic timezone changes, which means that the UTC offset used by `localtime` is not always the same as the value returned by `getTimezone()`, which returns the _current_ UTC offset. But `getTimezone()` is used to set the value of `TimeInfo.timezone`.

Code which reproduces the issue in the Asia/Novosibirsk timezone:

```nim
var lt = getLocaltime(fromSeconds(1469275200))
var gt = getGmTime(fromSeconds(1469275200))

echo "claimed timezone: ", lt.timezone # -25200
lt.timezone = 0
lt.isDst = false
echo "actual timezone: ", gt.toTime.int - lt.toTime.int # -21600
``` 

This inconsistency then causes other issues when using the TimeInfo. Note that the JS backend ignores historic offset changes, and will give `-21600` twice instead.

**Comment by andreaferretti** [comment]
@GULPF You seem to be the most expert about timezone handling, based on the issues you are opening. Would you consider helping improve the `times` module?

**Comment by GULPF** [comment]
I wouldn't say I'm an expert, but I enjoy dealing with date & time issues. I've already done a few PR's for the `times` module, and I'll continue with that.

I'm planning some larger changes in a few weeks when I have more time, mostly centered around adding proper timezone support.

**Comment by andreaferretti** [comment]
Great to hear that! :-)

**Comment by treeform** [comment]
I wrote a time/timezone library: https://github.com/treeform/chrono
It supports times from when we have good data bout them, maybe give it a try? Feature requests welcome.

**Comment by GULPF** [comment]
Fixed by #6552
